### PR TITLE
Fix `action` not set in the 'content-type' header in SOAP 1.2 

### PR DESCRIPTION
**Pull request by Nuvindu**
## Purpose
Fixes: [#5842](https://github.com/ballerina-platform/ballerina-library/issues/5842)

### CODE REVIEW

File: ballerina/modules/soap12/tests/http_soap_service.bal [hunk 1]
```diff
@@ -48,6 +48,20 @@ service / on new http:Listener(9090) {
         return response;
     }
 
+    resource function post getActionPayload(http:Request request) returns http:Response|error {
+        string[] headers = check request.getHeaders(mime:CONTENT_TYPE);
+        mime:MediaType mediaHeader = check mime:getMediaType(headers[1]);
```
Comment (dilanSachi) [inline ballerina/modules/soap12/tests/http_soap_service.bal:53]: is it possible that the headers can be empty?

File: ballerina/soap_utils.bal [hunk 2]
```diff
@@ -150,16 +150,20 @@ isolated function createSoap11HttpRequest(xml|mime:Entity[] body, string soapAct
...
     http:Request req = new;
-    if body is xml {
-        req.setXmlPayload(body);
+    _ = body is xml ? req.setXmlPayload(body) : req.setBodyParts(body);
+    if soapAction is string {
+        map<string> stringMap = {};
+        stringMap["action"] = "\"" + soapAction + "\"";
+        var mediaType = mime:getMediaType(mime:APPLICATION_SOAP_XML);
+        if mediaType is mime:MediaType {
+            mediaType.parameters = stringMap;
```
Comment (dilanSachi) [inline ballerina/soap_utils.bal:159]:
```suggestion
            mediaType.parameters = {"action": string`"${soapAction}"`};

```

Comment (Nuvindu) [reply ballerina/soap_utils.bal]: Fixed in [703e5bb](https://github.com/ballerina-platform/module-ballerina-soap/pull/116/commits/703e5bbc039b4c7ec5d486e287916837506d7f7e)

**Comment by ThisaruGuruge** [comment]
Since this is a backword-incompatible change, let's update the version to next minor version.

**Review by ThisaruGuruge — APPROVED** [review]
LGTM

**Pull request merged**